### PR TITLE
[Type Hints] DDIM pipelines

### PR DESCRIPTION
**Pull request by sidthekidder**
Adding type hints for 30, 31 and 32 from #287.
I referenced [this code](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L18) while adding types for `LDMTextToImagePipeline.__init__`, but wasn't sure of the right classes - they could be too broad.

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]
That looks great to me!

**Review by anton-l — APPROVED** [review]
LGTM, thanks @sidthekidder!

File: src/diffusers/pipelines/ddim/pipeline_ddim.py [hunk 1]
```diff
@@ -31,12 +31,11 @@ def __init__(self, unet, scheduler):
...
-        generator=None,
-        eta=0.0,
-        num_inference_steps=50,
-        output_type="pil",
-        return_dict: bool = True,
+        batch_size: int = 1,
+        generator: Optional[torch.Generator] = None,
+        eta: float = 0.0,
+        num_inference_steps: int = 50,
+        output_type: Optional[str] = "pil",
```
Comment (anton-l) [inline src/diffusers/pipelines/ddim/pipeline_ddim.py:38]:
```suggestion
        output_type: Optional[str] = "pil",
        return_dict: bool = True,
```

File: src/diffusers/pipelines/ddpm/pipeline_ddpm.py [hunk 2]
```diff
@@ -30,7 +30,11 @@ def __init__(self, unet, scheduler):
 
     @torch.no_grad()
     def __call__(
-        self, batch_size=1, generator=None, output_type="pil", return_dict: bool = True, **kwargs
+        self,
+        batch_size: int = 1,
+        generator: Optional[torch.Generator] = None,
+        output_type: Optional[str] = "pil",
```
Comment (anton-l) [inline src/diffusers/pipelines/ddpm/pipeline_ddpm.py:36]:
```suggestion
        output_type: Optional[str] = "pil",
        return_dict: bool = True,
```

**Pull request merged**